### PR TITLE
Remove double dependencies in both devDependencies and dependencies

### PR DESCRIPTION
**Pull request by vazexqi**
### What does this PR do?
 
Also, remove dependencies from modules that do not need it.

### CODE REVIEW

File: packages/salesforcedx-vscode-apex/package.json [hunk 1]
```diff
@@ -23,7 +23,6 @@
   },
   "categories": ["Languages"],
   "devDependencies": {
-    "@salesforce/salesforcedx-utils-vscode": "40.4.0",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-apex/package.json]: @guw - Was there a reason you duplicated this in both devDependencies and dependencies?

File: packages/salesforcedx-vscode-lightning/package.json [hunk 2]
```diff
@@ -23,7 +23,6 @@
   },
   "categories": ["Languages"],
   "devDependencies": {
-    "@salesforce/salesforcedx-utils-vscode": "40.4.0",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-lightning/package.json]: @guw - Was there a reason you forced this one to have a devDependency on salesforcedx-utils-vscode even though it is not being used?

File: packages/salesforcedx-vscode-visualforce/package.json [hunk 3]
```diff
@@ -23,7 +23,6 @@
   },
   "categories": ["Languages"],
   "devDependencies": {
-    "@salesforce/salesforcedx-utils-vscode": "40.4.0",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-visualforce/package.json]: @guw - Was there a reason you forced this one to have a devDependency on salesforcedx-utils-vscode even though it is not being used?

**Comment by vazexqi** [comment]
OK. It seems that you moved the testrunner stuff into the salesforcedx-utils-vscode and now everything needs to depend on that module.

**Comment by vazexqi** [comment]
Travis CI seems to be stuck on the OS X builds. 
I'm by passing this so that we can get develop into a good building state.

**Comment by guw** [comment]
@vazexqi Is there a way to let lerna handle the versioning automatically? 

FWIW, I wasn't if devDependencies would be sufficient. It's a build/test dependency only and not a full runtime dependency.

**Pull request merged**